front/lib/api/assistant/conversation.ts

```diff
@@ -1655,3 +1669,39 @@ async function* streamRunAgentEvents(
     }
   }
 }
+
+async function isMessagesLimitReached({
+  owner,
+  plan,
+}: {
+  owner: WorkspaceType;
+  plan: PlanType;
+}): Promise<boolean> {
+  if (plan.limits.assistant.maxMessages === -1) {
+    return false;
+  }
+  /** Counting on workspace because user can be null */
+  const count = await Conversation.count({
+    include: [
+      {
+        model: Message,
+        required: true,
```

Comment:
If `required: true` this is an inner join no? so that counts only the Conversation?